app/controllers/observations/maps_controller.rb

```diff
@@ -42,19 +43,24 @@ def find_locations_matching_observations
         limit: 10_000
       }
       @observations =
-        @query.select_rows(args).map do |id, lat, long, gps_hidden, loc_id|
-          locations[loc_id.to_i] = nil if loc_id.present?
-          lat = long = nil if gps_hidden == 1
-          Mappable::MinimalObservation.new(id, lat, long, loc_id)
+        @query.select_all(args).map do |obs|
+          obs.symbolize_keys!
+          # Setting this to nil is a flag to look up the location. Because we
+          # selected obs attributes not instances, we can't call `obs.location`
+          # and we shouldn't anyway. Getting the Locations in bulk is quicker.
+          locations[obs[:location_id].to_i] = nil if obs[:location_id].present?
```

Comment:
Minor point, but I think it would clearer if in this function `locations` was `location_ids` and handled with a Set and then `eager_load_corresponding_locations` returned the dictionary.  As written `locations` is kind of playing a double role and getting updated as a side effect in `eager_load_corresponding_locations`.

Reply:
@mo-nathan Can you show me what you mean?

Reply:
Reading up on Ruby `Set`. I don't remember ever coming across this class, incredibly.

Reply:
Hmm, why don't we use a Set in the name validation refactor?

Reply:
I was in favor of using a Set when I thought we had to do things iteratively.  Now that we can get all the data in a single SQL query, it's not really worth it.  In general Sets are great.  They are way more efficient than lists or dictionary and folks are often aware of them or don't think to use them.